File: 프로그래머스/unrated/181935. 홀짝에 따라 다른 값 반환하기/홀짝에 따라 다른 값 반환하기.js
function solution(n) {
    let sum = 0;
    let i = n % 2 === 0 ? 2 : 1;
    
    for (i; i <= n; i += 2) {
        if (n % 2 === 0) {
            sum += i ** 2;
        } else {
            sum += i;
        }
    }
    return sum;
}

File: 프로그래머스/unrated/181935. 홀짝에 따라 다른 값 반환하기/홀짝에 따라 다른 값 반환하기.java
class Solution {
    public int solution(int n) {
        int sum = 0;
        boolean isEven = n % 2 == 0;
        
        if (isEven) {
            for (int i = 2; i <= n; i += 2) {
                sum += i * i;
            }
        } else {
            for (int i = 1; i <= n; i += 2) {
                sum += i;
            }
        }
        
        return sum;
    }
}